WebRoot/js/zimbraMail/mail/view/prefs/ZmAccountsPage.js

```diff
@@ -318,6 +318,381 @@ function(params) {
 	this._oneTimeCodesDialog.popup();
 };
 
+/*
+ * Recovery Account Section
+ */
+
+/*
+ * Validate Button's Listener.
+ *
+ * Validate code input through the SetRecoveryAccountRequest API Method.
+ *
+ */
+ZmAccountsPage.prototype._handleValidateRecoveryEmailButton =
+	function() {
+	var soapDoc = AjxSoapDoc.create("SetRecoveryAccountRequest", "urn:zimbraMail");
+	var callback = this._recoveryValidateRequest.bind(this);
+	var errorCallback = this._recoveryErrorCallback.bind(this);
+	this.recoveryErrorMessage.innerHTML = "";
+	soapDoc.setMethodAttribute("op", "validateCode");
+	soapDoc.setMethodAttribute("recoveryAccountVerificationCode", this.recoveryCodeInputField.getValue());
+	appCtxt.getAppController().sendRequest({soapDoc: soapDoc, asyncMode: true, callback: callback, errorCallback: errorCallback});
+};
+
+/*
+ * Validate listener callback method.
+ *
+ * Expecting a blank response so the callback acts as confirmation that the code is verified.
+ * Clear the code input field, reset the button, update controls based on "verified" status.
+ *
+ * @param {object} response The response object.
+ */
+ZmAccountsPage.prototype._recoveryValidateRequest =
+function(response) {
+	this.recoveryCodeInputField.setValue("", true);
+	if(!response._isException) {
+		this.recoveryCodeInputField.setValue("", true);
+		this.validateRecoveryEmailButton.setEnabled(false);
+		this._setAccountPasswordControls(this.RECOVERY_EMAIL_STATUS.verified);
+	} else {
+		this._recoveryErrorCallback({code: "Unknown", msg: "Unknown Error: " + ZmMsg.unknownError});
+	}
+};
+
+/*
+ * Reset Button's Listener.
+ *
+ * Reset code and email input through the SetRecoveryAccountRequest API Method.
+ *
+ */
+ZmAccountsPage.prototype._handleResetRecoveryEmailButton =
+	function() {
+	var soapDoc = AjxSoapDoc.create("SetRecoveryAccountRequest", "urn:zimbraMail");
+	var callback = this._recoveryResetRequest.bind(this);
+	var errorCallback = this._recoveryErrorCallback.bind(this);
+	this.recoveryErrorMessage.innerHTML = "";
+	soapDoc.setMethodAttribute("op", "reset");
+	appCtxt.getAppController().sendRequest({soapDoc: soapDoc, asyncMode: true, callback: callback, errorCallback: errorCallback});
+};
+
+/*
+ * Reset listener callback method.
+ *
+ * Expecting a blank response so the callback acts as confirmation that the code and email has been reset.
+ * Clear the code and email input fields, reset the buttons, update controls based on "null" status.
+ *
+ * @TODO use this.recoveryCodeValidity to manage code verification
+ * @TODO use this.recoveryResetExpiry to enable / disable reset button - might need a timer?
+ * @param {object} response The response object.
+ */
+ZmAccountsPage.prototype._recoveryResetRequest =
+function(response) {
```

Comment:
Previously we tabbed `function`, maybe just move them all to the same line as the declaration? Consider this for the rest of these method declarations.

Reply:
Im adhering to what was in all the other scripts - I won't say I am a fan, but it seems to be the norm here.

Reply:
Im adhering to what was in all the other scripts - I won't say I am a fan, but it seems to be the norm here.

Reply:
See line 369 for comparison to understand what I mean.